mdata/aggmetric.go

```diff
@@ -503,13 +429,13 @@ func (a *AggMetric) GC(chunkMinTs, metricMinTs uint32) bool {
 		return false
 	}
 
-	if currentChunk.LastWrite < chunkMinTs {
-		if currentChunk.Saved {
+	if a.lastWrite < chunkMinTs {
+		if a.lastSaveStart >= currentChunk.T0 {
 			// already saved. lets check if we should just delete the metric from memory.
-			if currentChunk.LastWrite < metricMinTs {
+			if a.lastWrite < metricMinTs {
 				return true
 			}
-		} else if !currentChunk.Saving {
+		} else if a.lastSaveStart < currentChunk.T0 {
 			// chunk hasn't been written to in a while, and is not yet queued to persist. Let's persist it
 			log.Debug("Found stale Chunk, persisting it to Cassandra. key: %s T0: %d", a.Key, currentChunk.T0)
 			a.persist(a.CurrentChunkPos)
```

Comment:
The chunk needs to be "Finished" before calling persist and persist should only be called if the node is a primary